app/src/oculusvr/cpp/DeviceDelegateOculusVR.cpp

```diff
@@ -674,6 +674,9 @@ struct DeviceDelegateOculusVR::State {
   uint32_t frameIndex = 0;
   double predictedDisplayTime = 0;
   ovrTracking2 predictedTracking = {};
+  ovrTracking2 discardPredictedTracking = {};
+  uint32_t discardedFrameIndex = 0;
+  int discardCount = 0;
```

Comment:
nit: Please unify these variables' naming, `discarded` or `discard`? I would consider to replace `discardFrameIndex` with `discardFrameIndex`.